apps/mac.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+
+#include "apps.h"
+#include "progs.h"
+#include <openssl/bio.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+#undef BUFSIZE
+#define BUFSIZE 1024*8
+
+typedef enum OPTION_choice {
+    OPT_ERR = -1, OPT_EOF = 0, OPT_HELP,
+    OPT_MAC, OPT_MACOPT, OPT_BIN, OPT_OUT
+} OPTION_CHOICE;
+
+const OPTIONS mac_options[] = {
+    {OPT_HELP_STR, 1, '-', "Usage: %s [options] [file]\n"},
+    {OPT_HELP_STR, 1, '-', "  file\t\tto mac (default is stdin)\n"},
+    {"help", OPT_HELP, '-', "Display this summary"},
+    {"name", OPT_MAC, 's', "MAC algorithm (See list -mac-algorithms)"},
+    {"ctrl", OPT_MACOPT, 's', "MAC algorithm control parameters in n:v form. "
```

Comment:
I think it would be more helpful to follow the example of the `pkey` commands and name that option `-macopt`